include/uapi/linux/cartesi/rollup.h

```diff
@@ -1,155 +1,74 @@
- *  - Yields automatic with tx-voucher
- *  - Fills in the index field with the corresponding slot from voucher-hashes
- * on success:
- *   Returns 0
- * on failure:
- *   EOPNOTSUPP in case the driver is currently processing an inspect state
- *   EFAULT in case of invalid arguments
- *   ERESTARTSYS in case of an internal lock error
- *   EDOM in case of an integer larger than 64bits is received
- *   EIO in case of yield device error */
-#define IOCTL_ROLLUP_WRITE_VOUCHER _IOWR(0xd3, 1, struct rollup_voucher)
-
-/* Outputs a new notice.
- * How: Computes the Keccak-256 hash of payload and then, atomically:
- *  - Copies the (be32(0x20)+be32(payload_length)+payload) to the tx buffer
- *  - Copies the hash to the next available slot in the notice-hashes memory range
- *  - Yields automatic with tx-notice
- *  - Fills in the index field with the corresponding slot from notice-hashes
- * on success:
- *   Returns 0
- * on failure:
- *   EOPNOTSUPP in case the driver is currently processing an inspect state
- *   EFAULT in case of invalid arguments
- *   ERESTARTSYS in case of an internal lock error
- *   EDOM in case of an integer larger than 64bits is received
- *   EIO in case of yield device error */
-#define IOCTL_ROLLUP_WRITE_NOTICE  _IOWR(0xd3, 2, struct rollup_notice)
-
-/* Outputs a new report.
- *  - Copies the (be32(0x20)+be32(payload_length)+payload) to the tx buffer
- *  - Yields automatic with tx-report
- * on success:
- *   Returns 0
- * on failure:
- *   EFAULT in case of invalid arguments
- *   ERESTARTSYS in case of an internal lock error
- *   EIO in case of yield device error */
-#define IOCTL_ROLLUP_WRITE_REPORT  _IOWR(0xd3, 3, struct rollup_report)
-
-/* Throws an exeption.
- *  - Copies the (be32(0x20)+be32(payload_length)+payload) to the tx buffer
- *  - Yields manual with tx-exception
- * on success:
- *   Returns 0
- * on failure:
- *   EFAULT in case of invalid arguments
- *   ERESTARTSYS in case of an internal lock error
- *   EIO in case of yield device error */
-#define IOCTL_ROLLUP_THROW_EXCEPTION  _IOWR(0xd3, 4, struct rollup_exception)
+/** Return a @p rollup_setup structure filled with the rollup buffer details
+ * (tx and rx). Use the values to mmap @p tx and @p rx into the user address
+ * space.
+ *
+ * @return
+ *   0      - success
+ *   EIO    - device error
+ *   EFAULT - invalid arguments */
+#define IOCTL_ROLLUP_SETUP                   _IOR (0xd3, 0, struct rollup_setup)
+
+/** Let the driver know userspace successfully processed the current input and
+ * will wait for the next. Upon ioctl return, the driver will fill the input
+ * size in @p argp and return the input type.
+ *
+ * @return
+ *   CARTESI_ROLLUP_ADVANCE_STATE - success, advance state was requested
+ *   CARTESI_ROLLUP_INSPECT_STATE - success, inspect state was requested
+ *   CARTESI_ROLLUP_INVALID_STATE - failure, unknown request type and/or error */
+#define IOCTL_ROLLUP_WAIT_FOR_INPUT          _IOR (0xd3, 1, __u64)
```

Comment:
The documentation of this ioctl is missing possible errors (all other ioctl shows them).